### ISSUE TITLE
Add New Antiblock Variant (And Recipe)

### ISSUE THREAD

**Issue by Drullkus** [TheCricket/Chisel-2#138]
We should add a new Antiblock variant where the lines will connect regardless of color. Render the black line CTM on a different layer?

Thankfully since we got a new GUI, we can stack this on top of the other Antiblock chisel list.

Also, localization should be changed to Antiblock and it needs a recipe too.

**Comment by tterrag1098** [comment]
Rendering the CTM on a different layer wouldn't be needed, it would just require a subclass of the existing RenderBlocks to not check metadata when connecting.

**Comment by CptSpaceToaster** [comment]
Question: Why does that logic take place in the renderer?  From a 30 second thought process, it would make a tiny bit more sense for the <Can X block connect to Y block> logic to be moved somewhere else...

For now, yes... We can remove the check on metadata for now... but I'm not 100% certain that we need an entire new set of 16 Antiblocks, that effectivly look the same, minus a couple black borders.

If anything... we want this configuration in resources.  Each CTM block could have a list of one-way connections.  Glass would connect to not only itself, but colored glass, etc.  Antiblocks could have a "ignore metadata" option enabled.  

That way, one set of textures (with a black border) could be configured by the end user, using a text file somewhere. 

Thoughts?  That got huge... but the framework sounds somewhat useful.

**Comment by warlordjones** [comment]
Options restrict the user to one or the other though, whereas new blocks allows use of them side by side. It was the same with having a config option for animated and non-animated voidstone, animated voidstone works much better in small amounts.

However, I quite like the idea of changing the framework to allow CTM connections to other blocks, that seems like it could be useful in the future, and may already have some uses (Technical Block rusty and non-rusty grate ctming together, for example?)

So, I'd support changing the framework, perhaps updating some existing blocks to support the change, and also support some configuration by the end user, but also think that adding a new block in this case could be justified. Indeed, some people may dislike all colours of glass connecting via CTM, for example, as I've seen it used effectively for patterns, and connection may spoil the effect a little. I think with the new chisel GUI, there is no reason not to allow things like this side by side. After all, more options is always good!

TL;DR Like the idea of allowing more flexibility in the framework but don't quite like the idea of individually configurable options, as I prefer allowing more side-by-side options.

**Comment by CptSpaceToaster** [comment]
I'd hate to proliferate the already bloated block list with BLOCK_NAME<Will Always Connect> and a seemingly identical BLOCK_NAME<Listens To Connection Configuration>

To reduce the amount of block duplication that's being asked for, moving this configuration to the end user sounds like the most ideal solution.  It's a lot of work, and a lot of framework would need to be written in order to do this saftly, but caching the blocks connection config, would allow different types of metals, stone, glass, antiblocks, etc, to be configured individually.  

Instead of asking ourselves: Do we want marble CTM textures to connect to each other? or do we want Antiblock CTM textures to connect to each other?  All we have to say is, configure it how you want!  And choose carefully!

This removes seemingly identical looking blocks that would have similar names from showing up in game.  I see that being quite confusing to most users.  Even if slightly more practical, it simply bloats up the UI, and makes recipes a nightmare.  (how do users know which Antiblock recipe to build, etc)

**Comment by Drullkus** [comment]
I was talking about adding Antiblock2 list to the Antiblock(1) list.

 Remember,   We have 60 slots now. That's waaay more than what we should have but hell, why not 60 slots.

One thing though is that I can add a 30-block GUI that we can switch to if blockCount =< 30.

**Comment by Drullkus** [comment]
Oh, just read the rest.

I think removing the CTM meta on the Antiblocks is a good idea.

**Comment by CptSpaceToaster** [comment]
Because antiblocks have their own personal renderer right now, and because no one else is using it, we can patch this incredibly easily by disabling the check for meta in that renderer.  **This is only a temporary solution**, and it would be good to consider a system that will deal with this problem in the future.

Your opinion is important here, as you work a lot with our textures, and I would love if you could weigh in on the conversation a bit more.  **For example:** There was some talk about a bunch of metal-like blocks.  If we wanted to add CTM to some of them, but not all, we might run into a couple issues:
- The new blocks might not all have the same block ID, making our hack not work well.
- We would be disabling the check for metadata blocks for **any other block that uses that renderer**.  Marble would start behaving differently, unless we duplicated the renderer again... and that's not a good solution everytime.  (It was necessary for Antiblocks, because they needed to render differently in the first place)
There are two discussed solutions so far.

1.) We make two renderers.  One would always connect CTM to related metadata blocks, and one would only allow the block to connect to itself.  Then, we have to make a metal block that has "ignore metadata", and a separate metal block that "pays attention to metadata" when rendering.

**Pros:** 
- Users can use each type of marble block, and they are not restricted.  ie, If they want use CTM metal that connects to other metadata-metal in one place, and they want to stop CTM in another, they can do so, simply by using the respective block.

**Cons:**
- Duplicates several blocks in Chisel, as the blocks are *incredibly similar* / User might be confused when they see two versions of metal all of a sudden.  Do we stop at metal?  What about Marble?  Others?
- Code would bloat, as blocks and renderers are duplicated
- Recipes would become more confusing. (do you want the connecting block, or the non-connecting version?)
- Blocks can ONLY connect to blocks with the same block ID, and the only difference is whether metadata is checked.

2.) We build a large system (probably using text files) in the resources folder, that controls which blocks connect to which when using CTM.  The current renderers would have to look at the new information, and render accordingly

**Pros:** 
- Any block can connect to any block, all based on configuration.
- Metadata limitations are removed.  
- No need to duplicate blocks in code (which solves recipe confusion)

**Cons:** 
- Blocks can only connect, or not connect... not both.  Users will lose a bit of flexibility, and have to choose outright, (connect or no) in the text-configuration.
- The required system necessary would take a while to create.  
- Requires more end-user configuration.  Also, I don't know how to handle "two-way connections". ie,  If I wanted glass to connect to marble... would marble ALSO have to connect to glass?  (I think that can be solved in code, but it has to be handled carefully)


*There might be an option 3 here... but that's what I have so far.*
![TLDR](https://s-media-cache-ak0.pinimg.com/originals/1d/99/d7/1d99d7f286fec6bac7394be6c44d734a.jpg) We can change stuff, but we have to be careful how we change it.

**Comment by Drullkus** [comment]
What if we did Choice 1, but with this extension

Basically we add a new tool that can 'switch' a block between two diff blocks.

Let's say, Grates and Rusty Grates.

The old ones will not connect but you can use the tool in the world to make them connect. Behind the curtains we do this by adding a second Grate block that ignores meta CTM.

**Comment by Drullkus** [comment]
Hit the wrong button -_-

**Comment by tterrag1098** [comment]
That...that gif....

**Comment by Drullkus** [comment]
IKR... xD
